src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/kafka/KafkaOppfolgingstilfellePersonService.kt

```diff
@@ -1,18 +1,79 @@
 package no.nav.syfo.oppfolgingstilfelle.kafka
 
+import no.nav.syfo.application.database.DatabaseInterface
 import no.nav.syfo.application.kafka.KafkaConsumerService
+import no.nav.syfo.oppfolgingstilfelle.passererAktivitetskravVurderingStoppunkt
 import org.apache.kafka.clients.consumer.*
+import org.slf4j.LoggerFactory
+import java.sql.Connection
 import java.time.Duration
 
-class KafkaOppfolgingstilfellePersonService : KafkaConsumerService<KafkaOppfolgingstilfellePerson> {
+class KafkaOppfolgingstilfellePersonService(val database: DatabaseInterface) :
+    KafkaConsumerService<KafkaOppfolgingstilfellePerson> {
 
     override val pollDurationInMillis: Long = 1000
 
     override fun pollAndProcessRecords(kafkaConsumer: KafkaConsumer<String, KafkaOppfolgingstilfellePerson>) {
         val records = kafkaConsumer.poll(Duration.ofMillis(pollDurationInMillis))
         if (records.count() > 0) {
-            // TODO: Process records
+            processRecords(records)
             kafkaConsumer.commitSync()
         }
     }
+
+    private fun processRecords(records: ConsumerRecords<String, KafkaOppfolgingstilfellePerson>) {
+        val (tombstoneRecords, validRecords) = records.partition { it.value() == null }
+        processTombstoneRecords(tombstoneRecords)
+        processValidRecords(validRecords)
+    }
+
+    private fun processTombstoneRecords(tombstoneRecords: List<ConsumerRecord<String, KafkaOppfolgingstilfellePerson>>) {
+        if (tombstoneRecords.isNotEmpty()) {
+            val numberOfTombstones = tombstoneRecords.size
+            log.error("Value of $numberOfTombstones ConsumerRecords are null, most probably due to a tombstone. Contact the owner of the topic if an error is suspected")
+            COUNT_KAFKA_CONSUMER_OPPFOLGINGSTILFELLE_PERSON_TOMBSTONE.increment(numberOfTombstones.toDouble())
+        }
+    }
```

Comment:
For en med lite kafka-erfaring og som bare har hørt ordet, hva er en egentlig tombstone record?

Reply:
Leste litt her: https://medium.com/@damienthomlutz/deleting-records-in-kafka-aka-tombstones-651114655a16 👍🏼